packages/front-end/pages/environments.tsx

```diff
@@ -227,13 +228,14 @@ const EnvironmentsPage: FC = () => {
                               color=""
                               className="dropdown-item"
                               onClick={async () => {
-                                const newEnvs = [...environments];
-                                newEnvs.splice(i, 1);
-                                newEnvs.splice(i + 1, 0, e);
+                                const index = environments.findIndex(
+                                  (env) => env.id === e.id
+                                );
                                 await apiCall(`/environment/order`, {
                                   method: "PUT",
                                   body: JSON.stringify({
-                                    environments: newEnvs.map((env) => env.id),
+                                    envId: e.id,
+                                    newIndex: index + 1,
```

Comment:
should actually get the global index of `filteredEnvironments[i+1]`

Reply:
If we change this, we also need to change the render condition for this button.  It currently does:

```ts
i < environments.length - 1 &&
```